src/ParallelAlgorithms/LinearSolver/AsynchronousSolvers/ElementActions.hpp

```diff
@@ -163,24 +190,61 @@ struct InitializeElement {
   }
 };
 
-template <typename OptionsGroup>
+template <typename OptionsGroup, typename ArraySectionIdTag = void>
 struct RegisterObservers {
   template <typename ParallelComponent, typename DbTagsList,
             typename ArrayIndex>
   static std::pair<observers::TypeOfObservation, observers::ObservationKey>
-  register_info(const db::DataBox<DbTagsList>& /*box*/,
+  register_info(const db::DataBox<DbTagsList>& box,
                 const ArrayIndex& /*array_index*/) noexcept {
+    // Get the observation key, or "__none" if the element does not belong
+    // to a section with this tag. In the latter case, no observations will
+    // ever be contributed.
+    const std::string section_observation_key =
+        get_section_observation_key<ArraySectionIdTag>(box).value_or("__none");
```

Comment:
What about replacing `__none` with `DoNotPerformReduction`? While not a C++ entity, `__` prefixes in C++ are reserved for compilers and STL, so I'm a little worried that such a string will encourage a variable named `__none` :Shrug:

Reply:
I went with "Unused" now. My intention was to make sure this special value doesn't conflict with anything that developers might use as a section key, but "Unused" is probably pretty safe. Added an assert to be sure.